omod/src/main/webapp/resources/scripts/conditionlist/controllers/editcondition.controller.js

```diff
@@ -0,0 +1,147 @@
+    var self = this;
+
+    $scope.patientUuid = null;
+    $scope.condition = null;
+    const INACTIVE_STATUS = 'Inactive';
+
+    // this is required inorder to initialize the Restangular service
+    RestfulService.setBaseUrl('/' + OPENMRS_CONTEXT_PATH + '/ws/rest/emrapi');
+
+    /**
+     * Perform POST save
+     * PS: Make sure the 'Non Coded Uuid' setting (under Settings->Condition List) is set inorder to work with noncoded concepts,
+     * the LOCALE needs to be set (most concepts have 'en')
+     * @type {Function}
+     */
+    self.saveCondition = self.saveCondition || function () {
+        if($scope.condition.onSetDate>$scope.condition.endDate){
+                emr.errorAlert("End Date can't be before Onset Date");
+        }
+        else{
+            var conditions = [];
+            conditions.push($scope.condition);
+            RestfulService.post('condition', conditions, function (data) {
+                emr.successAlert("Condition edited successfully")
+                window.location = '/' + OPENMRS_CONTEXT_PATH + '/coreapps/conditionlist/manageConditions.page?patientId=' + $scope.patientUuid + '&';
+            }, function (error) {
+                emr.errorAlert("Error Saving condition");
+            });
+        }
+    }
+
+    self.initCondition = self.initCondition || function () {
+            $scope.patientUuid = CommonFunctions.extractUrlArgs(window.location.search)['patientId'];
+            $scope.condition = new ConditionModel($scope.patientUuid)
+            $scope.condition = CommonFunctions.extractUrlArgs(window.location.search)['condition'];
+            self.convertURLParameterToJSON();
+
+            $scope.concept = {
+                display: $scope.condition.concept.name,
+                concept: {
+                    display: $scope.condition.concept.name,
+                },
+                conceptName: {
+                    display: $scope.condition.concept.name,
+                }
+            }
+
+            var onsetDatePicker = angular.element(document.getElementsByName('conditionStartDate'))[0];     
+            onsetDatePicker.value = self.convertForDateValue($scope.condition.onSetDate);
+          
+            if($scope.condition.status==INACTIVE_STATUS){
+                var endDatePicker = angular.element(document.getElementsByName('conditionEndDate'))[0];      
+                endDatePicker.value = self.convertForDateValue($scope.condition.endDate);
+            } 
+
+            var inputs = document.getElementsByTagName('input');
+            inputs[0].disabled = true;
+            inputs[1].value = self.convertToLongDate($scope.condition.onSetDate);  
+            self.showEndDate();
+        }
+    
+    self.convertURLParameterToJSON = self.convertURLParameterToJSON || function(){    
+        $scope.condition = JSON.parse($scope.condition.split("%22").join("\"").split("%20").join(" "));
```

Comment:
Can you give an example of what the input url parameter could look like and what the converted value would be?

Reply:
Input url parameter would look something like -  

{%22uuid%22:%222711b0d7-8bea-4f9e-832e-c4c6a5b3fbab%22,%22patientUuid%22:%22e26f6ab6-df77-4d52-9b37-1f231988a3bb%22,%22concept%22:{%22uuid%22:%22119027AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA%22,%22name%22:%22Disease%20of%20bone%20and%20joint%22,%22shortName%22:null},%22conditionNonCoded%22:%22%22,%22status%22:%22ACTIVE%22,%22onSetDate%22:1561939200000,%22endDate%22:null,%22endReason%22:null,%22additionalDetail%22:%22%22,%22voided%22:false,%22voidReason%22:null,%22creator%22:%22Super%20User%20(null)%22,%22dateCreated%22:1562188446000,%22previousConditionUuid%22:null}

Converted value would look something like - 

{
  "uuid": "2711b0d7-8bea-4f9e-832e-c4c6a5b3fbab",
  "patientUuid": "e26f6ab6-df77-4d52-9b37-1f231988a3bb",
  "concept": {
    "uuid": "119027AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA",
    "name": "Disease of bone and joint",
    "shortName": null
  },
  "conditionNonCoded": "",
  "status": "ACTIVE",
  "onSetDate": 1561939200000,
  "endDate": null,
  "endReason": null,
  "additionalDetail": "",
  "voided": false,
  "voidReason": null,
  "creator": "Super User (null)",
  "dateCreated": 1562188446000,
  "previousConditionUuid": null
}

Reply:
Can you point me to where the original string is constructed from?

Reply:
It is automatically being constructed by the browser as a part of url.

Reply:
Which page should i access to construct it?

Reply:
I'm using redirectToEditCondition function in manageconditions.controller to pass the information of condition as a part of url. The browser is automatically encoding double quotes to %22 and space to %20.

Reply:
Is there any reason why you did not pass this as a POST request's body, instead of url?

Reply:
I had no idea how post could be used in OpenMRS. Could you please provide some resources and guidance on this :)

Reply:
Just a normal HTTP post. Nothing specific to OpenMRS

Reply:
How far are you with this?

Reply:
I have been encountering a few error's (specially GroovyRuntimeException) while using POST. Also POST request can be sent through a form, but how do i retrieve data from it into the edit conditions page? Do I need to use the EditConditionPageController.java ? Could you provide me with an example of how to retrieve POST data?